trunk/src/main/java/cn/edu/uestc/acmicpc/service/impl/ProblemServiceImpl.java

```diff
@@ -40,6 +41,16 @@ public ProblemServiceImpl(ProblemDao problemDao) {
         problemCondition.getCondition());
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  public List<Integer> getAllVisibleNormalProblemIds() throws AppException {
```

Comment:
I prefer merge all usages into a method, i.e. set visible and type as parameters.